File: addbinary.py
# Link to the leetcode problem
# https://leetcode.com/problems/add-binary/
# I am attempting this problem in python to avoid bit limitation of 
# C++ since the constraints of this problem is set whereby
#  each string input could represent an integer that is
# 1000 bits long

class Solution:
	def addBinary(self, a: str, b: str) -> str:
		one = 0
		two = 0
		for i in range(0, len(a)):
			one <<= 1
			if (a[i] == '1'):
				one += 1
		for i in range(0, len(b)):
			two <<= 1
			if (b[i] == '1'):
				two += 1
		sum = one + two
		if (sum == 0):
			return ("0")
		# print(one)
		# print(two)
		# print(sum)
		result = ""
		while (sum != 0):
			if (sum & 1 == 1):
				result = "1" + result
			else:
				result = "0" + result
			sum >>= 1
		return result

s1 = "11"
s2 = "1"
s = Solution()
s3 = s.addBinary(s1, s2)
print(s3)

File: addbinary.cpp
/*
Link to the leetcode problem
https://leetcode.com/problems/add-binary/
*/

#include <iostream>
#include <vector>
#include <algorithm>
#include <string>

using std::vector;
using std::cout;
using std::endl;
using std::string;
using std::reverse;

class Solution {
public:
    string addBinary(string a, string b)
	{
		int i = a.length();
		int j = b.length();
		string result;
		while (i >= 0 || j >= 0)
		{
			if (a[i] == '1' && b[j] == '1')
				result += '10';
		}
		
    }
};

void	runcode(void)
{
	string s2 = "10010101110001";
	string s1 = "101100011010111001101";
	Solution s;
	cout << s.addBinary(s1, s2) << endl;
}

int	main(void)
{
	runcode();
	return (0);
}